client/internal/statemanager/manager.go

```diff
@@ -0,0 +1,298 @@
+package statemanager
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/exp/maps"
+
+	nberrors "github.com/netbirdio/netbird/client/errors"
+)
+
+// State interface defines the methods that all state types must implement
+type State interface {
+	Name() string
```

Comment:
If accidentally implement multiple states with the same name then it can cause an issue. Maybe would be good to maintain the names in this pkg.

Reply:
I could add a check in `RegisterState` but that  means we need to return an `error` which adds more refactoring/boilerplate. Is it worth it?

Reply:
I am just thinking in exported const string what will be used by the implementation of the interface.